oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/DefaultOAuth2AuthorizationRequestResolver.java

```diff
@@ -201,6 +207,28 @@ private static String expandRedirectUri(HttpServletRequest request, ClientRegist
 				.toUriString();
 	}
 
+	/**
+	 * Creates nonce and its hash for use in OpenID Connect Authentication Requests
+	 *
+	 * @param attributes where {@link IdTokenClaimNames#NONCE} is stored for the token request
+	 * @param additionalParameters where hash of {@link IdTokenClaimNames#NONCE} is added to the authentication request
+	 *
+	 * @since 5.2
+	 * @see <a target="_blank" href="https://openid.net/specs/openid-connect-core-1_0.html#NonceNotes">15.5.2.  Nonce Implementation Notes</a>
+	 * @see <a target="_blank" href="https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation">3.1.3.7.  ID Token Validation</a>
+	 */
+	private void addNonceParameters(Map<String, Object> attributes, Map<String, Object> additionalParameters) {
+		String nonce = this.stringKeyGenerator.generateKey();
+		attributes.put(IdTokenClaimNames.NONCE, nonce);
+
+		try {
+			String nonceHash = createHash(nonce);
+			additionalParameters.put(IdTokenClaimNames.NONCE, nonceHash);
```

Comment:
This should be using `OidcParameterNames.NONCE`

Reply:
Sorry about that, corrected.